subprojects/xcfa/xcfa-cli/src/main/java/hu/bme/mit/theta/xcfa/cli/portfolio/complex25.kt

```diff
@@ -0,0 +1,1137 @@
+import hu.bme.mit.theta.xcfa.cli.params.*
+import hu.bme.mit.theta.xcfa.cli.runConfig
+import hu.bme.mit.theta.xcfa.model.XCFA
+import hu.bme.mit.theta.xcfa.passes.*
+import java.nio.file.Paths
+
+fun complexPortfolio25(
+  xcfa: XCFA,
+  mcm: MCM,
+  parseContext: ParseContext,
+  portfolioConfig: XcfaConfig<*, *>,
+  logger: Logger,
+  uniqueLogger: Logger,
+): STM {
+
+  val checker = { config: XcfaConfig<*, *> -> runConfig(config, logger, uniqueLogger, true) }
+
+  var baseConfig =
+    XcfaConfig(
+      inputConfig =
+        InputConfig(
+          input = null,
+          xcfaWCtx = Triple(xcfa, mcm, parseContext),
+          propertyFile = null,
+          property = portfolioConfig.inputConfig.property,
+        ),
+      frontendConfig =
+        FrontendConfig(
+          lbeLevel = LbePass.level,
+          loopUnroll = LoopUnrollPass.UNROLL_LIMIT,
+          inputType = InputType.C,
+          specConfig = CFrontendConfig(arithmetic = ArchitectureConfig.ArithmeticType.efficient),
+        ),
+      backendConfig =
+        BackendConfig(
+          backend = Backend.CEGAR,
+          solverHome = portfolioConfig.backendConfig.solverHome,
+          timeoutMs = 0,
+          specConfig =
+            CegarConfig(
+              initPrec = InitPrec.EMPTY,
+              porLevel = POR.NOPOR,
+              porRandomSeed = -1,
+              coi = ConeOfInfluenceMode.NO_COI,
+              cexMonitor = CexMonitorOptions.CHECK,
+              abstractorConfig =
+                CegarAbstractorConfig(
+                  abstractionSolver = "Z3",
+                  validateAbstractionSolver = false,
+                  domain = Domain.EXPL,
+                  maxEnum = 1,
+                  search = Search.ERR,
+                ),
+              refinerConfig =
+                CegarRefinerConfig(
+                  refinementSolver = "Z3",
+                  validateRefinementSolver = false,
+                  refinement = Refinement.SEQ_ITP,
+                  exprSplitter = ExprSplitterOptions.WHOLE,
+                  pruneStrategy = PruneStrategy.FULL,
+                ),
+            ),
+        ),
+      outputConfig =
+        OutputConfig(
+          versionInfo = false,
+          resultFolder = Paths.get("./").toFile(), // cwd
+          cOutputConfig = COutputConfig(disable = true),
+          witnessConfig =
+            WitnessConfig(
+              disable = false,
+              concretizerSolver = "Z3",
+              validateConcretizerSolver = false,
+            ),
+          argConfig = ArgConfig(disable = true),
+          enableOutput = portfolioConfig.outputConfig.enableOutput,
+        ),
+      debugConfig = portfolioConfig.debugConfig,
+    )
+
+  val forceUnrolledXcfa =
+    xcfa.optimizeFurther(
+      listOf(
+        AssumeFalseRemovalPass(),
+        MutexToVarPass(),
+        AtomicReadsOneWritePass(),
+        LoopUnrollPass(2), // TODO: how much to force unroll?
```

Comment:
2 is okay, as it is only a first step and most errors are found with this limit anyway.